### DISCUSSION TITLE
Add custom button

### DISCUSSION THREAD

**Discussion by DKMiller71** [viresh-ratnakar/exolve#80] [Q&A] [answered]
Is there a way to have it add a custom button to the button set and link to a function?

For example: for a puzzle with specially marked cells, have a custom button "Get Marked Cells" that pops up the content of those cells?

I know this could be done in the preamble section or similar, but I'd like it be in the buttons group and follow the button formatting.

**Comment by viresh-ratnakar** [comment, marked as answer]
Sure, I've done it myself in [this puzzle](https://viresh-ratnakar.github.io/ribelle-gussalufz-1-solved.html). You do it from the `customizeExolve()` method that gets called automatically when using `createExolve()` (the `Exolve(...)` constructor also provides a parameter for such a function, if you're using the constructor directly).

```
function customizeExolve(puz) {
  const b = document.createElement('button');
  b.className = 'xlv-button';
  b.innerHTML = 'Dkm!';
  b.title = 'Click for Dkm!'
  b.addEventListener('click', (evt) => {
    alert('You clicked on Dkm!');
  });
  /**
   * You can append to puz.buttonRow1 if you prefer,
   * which you'd have to if the puzzle does not provide
   * solutions as there is no second row then.
   */
  puz.buttonRow2.appendChild(b);
}
```
